### PR TITLE
Move Default Options To Backend

### PR DESCRIPTION
**Pull request by mdingjan**
Extra issue.

Main change: defaults have been removed from the frontend html pages to the options.ts backend.
Also the implementation of the functionality of the 'Restore Options' button has therefore been changed.

Please review this PR by building the extension, trying to find any bugs in the frontend and of course reviewing the code. ;)

Quack!

### CODE REVIEW

File: src/octopeer-github/main/Options/Options.ts [hunk 1]
```diff
@@ -15,48 +15,55 @@ const Options = new (class Options {
...
+        [this.LOGGING]: [true, true],
+        [this.MOUSE_HOVER]: [true, true],
+        [this.MOUSE_CLICK]: [true, true],
+        [this.MOUSE_SCROLLING]: [true, true],
+        [this.MOUSE_POSITION]: [true, true],
+        [this.DATA_COMMENTS]: [true, true],
+        [this.DATA_KEYSTROKES]: [true, true],
+        [this.DATA_HTML]: [false, false],
+        [this.DATA_TABS]: [false, false],
+        [this.DATA_RESOLUTION]: [false, false],
```
Comment (mpsijm) [inline src/octopeer-github/main/Options/Options.ts]: I think we need to have a quick talk tomorrow about what will be the final default options :)

Comment (mdingjan) [inline src/octopeer-github/main/Options/Options.ts]: Sure, no problem. These options have just been extracted from the front end, so they won't be something new. But the overview makes it easier to change it, so yeeah, why not do this in this PR.

File: src/octopeer-github/main/Options/Options.ts [hunk 2]
```diff
@@ -15,48 +15,55 @@ const Options = new (class Options {
...
         this.observers = [];
-        let options: string[] = this.generateOptionList();
+        chrome.storage.sync.get("loggingEnabled", (res) => {
+            let options = this.generateOptionList();
+            if (res[this.LOGGING] === undefined) {
+                chrome.storage.sync.set(this.generateCurrentOptionMap());
+            } else {
+                chrome.storage.sync.get(options, (obj) => this.syncOptionMap(obj));
+            }
+            chrome.storage.local.set(this.generateDefaultOptionMap());
```
Comment (thervh70) [inline src/octopeer-github/main/Options/Options.ts]: Why local and not in sync storage?

**Comment by thervh70** [comment]
Looks good to me. This is ready to be merged.

File: src/octopeer-github/main/Options/Options.ts (hunk 2 above)
Comment (mdingjan) [inline src/octopeer-github/main/Options/Options.ts]: For functionality purposes. Default options can be stored in the local storage, as on reload of the extension, they won't change. It's nice to keep them separate for this reason.

File: resources/main.css [hunk 3]
```diff
@@ -18,19 +18,23 @@ p {
...
 
 input[type="checkbox"]:disabled+label:before {
-    opacity: 0;
     cursor: default;
+    font-size: 12pt;
+    opacity: 0;
 }
 
 input[type="checkbox"]+label:before {
+    font-size: 12pt;
```
Comment (mpsijm) [inline resources/main.css]: Can't the font-size be put in a more global definition like `body`?

Comment (mdingjan) [inline resources/main.css]: Yes, thanks!
Check the latest commit. :)

File: resources/options.html [hunk 4]
```diff
@@ -90,23 +97,28 @@
...
                                 </li>
                             </ul>
                         </div>
                     </div>
                 </div>
             </div>
-            <button class="btn waves-effect lime waves-light" type="submit" style="font-size: 11pt; margin-bottom: 30px;" id="restore">
+            <p style="font-size: 11pt;">
+                * If you do experience performance issues then we would advice you to first turn off HTML logging. Then if you
+                still experience performance issues, please turn off scroll logging. This should fix most issues on pretty
```
Comment (mpsijm) [inline resources/options.html]: pretty much -> most? it sounds a bit too informal.

Comment (mdingjan) [inline resources/options.html]: It says twice most then, but that okay.
Will do.

**Comment by mpsijm** [comment]
It's all looking so fancy now :D Can be merged! :)

File: resources/main.css [hunk 5]
```diff
@@ -33,5 +28,12 @@ input[type="checkbox"]:disabled+label:before {
 }
 
 input[type="checkbox"]+label:before {
-    transition(all .3s ease-in-out);
+transition(all .3s ease-in-out);
```
Comment (mpsijm) [inline resources/main.css]: Why remove the indentation here?

Comment (Exclaminator) [inline resources/main.css]: Exident, should be problemless now

**Comment by Exclaminator** [comment]
I fixed some styling issues, PR looks good, going to merge now.

**Pull request merged**